test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -618,13 +647,10 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 			})
 
 			ginkgo.By("Checking the workload remains admitted", func() {
-				gomega.Eventually(func() error {
-					err := k8sClient.Get(ctx, wlKey, updatedWl)
-					if err != nil {
-						return err
-					}
-					util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, string(updatedWl.Status.Admission.ClusterQueue), updatedWl)
-					return nil
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, updatedWl)
```

Comment:
No need to call `SyncAdmittedConditionForWorkloads` here. This function is used to transition from `QuotaReservation` to `Admitted` by scheduler (which is not present in the setup). 
However, here we don't expect this to be needed. One thing which could happen is that the workload_controller evicts the workload, but scheduler should not be involved.

Reply:
Correct.